etna/ciim/client.py

```diff
@@ -73,47 +73,40 @@ class Aggregation(str, enum.Enum):
     HELD_BY = "heldBy"
 
 
-def format_list_param(
-    items: Optional[list], param_type: Optional[str] = None
-) -> Optional[str]:
-    """Convenience function to transform list to comma-separated string.
-
-    When parsing a request's parameters `requests` uses `urllib.parse.urlencode`
-    with the deseq flag set to true, transforming a list's values into multiple
-    parameters:
-
-    >>> urllib.parse.urlencode({'param': ['item-1', 'item-2']}, doseq=True)
-    'param=item-1&param=item-2'
-
-    https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode
-
-    Kong expects multiple values to be a in a comma-separated string:
-
-    'param=item-1,+item-2'
-
-    This function, when given a list, will return a comma-separated string.
-
-    Additionally, to overcome an issue with the API, any commas within
-    paramter text will be removed.
+def prepare_filter_aggregations(items: Optional[list]) -> Optional[str]:
+    """
+    Filter format in items: 'field:value'
+    Prepares i.e. removes/replaces special chars from a filter fields' value to be passed to the api
+
+    Example:
+    before-prepare: "heldBy:Birmingham: Archives, Heritage and Photography Service"
+    after-prepare:  "heldBy:Birmingham  Archives  Heritage and Photography Service"
+    before-prepare: "heldBy:Staffordshire and Stoke-on-Trent Archive Service: Staffordshire County Record Office"
+    after-prepare:  "heldBy:Staffordshire and Stoke on Trent Archive Service  Staffordshire County Record Office"
+    Special char single quote i.e. ' is not prepared
+    before-prepare: "heldBy:Labour History Archive and Study Centre (People's History Museum/University of Central Lancashire)"
+    after-prepare:  "heldBy:Labour History Archive and Study Centre  People's History Museum University of Central Lancashire "
+    before-prepare: "heldBy:Bedfordshire Archives & Records Service"
+    after-prepare: "heldBy:Bedfordshire Archives   Records Service"
     """
     if not items:
         return None
 
-    if param_type == "filter_aggregations":
-        # reconstruct the required starting value that contains special value
-        # replace special chars by space
-        regex = r"([():\\/,&])"
-        subst = " "
-        items = [
-            i.split(":", 1)[0]
-            + ":"
-            + re.sub(regex, subst, i.split(":", 1)[1], 0, re.MULTILINE)
-            for i in items
-        ]
-    else:
-        items = [i.replace(",", "") for i in items]
+    regex = r"([/():,&-])"
+    subst = " "
+    field_list_to_prepare = ["heldBy"]
+    filter_prepared_list = []
+
+    for item in items:
+        split_field_value = item.split(":", 1)
+        field, value = split_field_value[0], split_field_value[1]
```

Comment:
You could shorten this to:

```suggestion
        field, value = item.split(":", 1)
```

Reply:
Done TY :)

Reply:
Done TY :)